pkg/cmd/codespace/create.go

```diff
@@ -111,6 +115,53 @@ func (a *App) Create(ctx context.Context, opts createOptions) error {
 		branch = repository.DefaultBranch
 	}
 
+	devContainerPath := opts.devContainerPath
+
+	// now that we have repo+branch, we can list available devcontainer.json files (if any)
+	if len(opts.devContainerPath) < 1 {
+		a.StartProgressIndicatorWithLabel("Fetching devcontainer.json files")
+		devcontainers, err := a.apiClient.ListDevContainers(ctx, repository.ID, branch, 100)
+		if err != nil {
+			return fmt.Errorf("error getting devcontainer.json paths: %w", err)
+		}
+		a.StopProgressIndicator()
+
+		if len(devcontainers) > 0 {
+
+			// if there is only one devcontainer.json file and it is one of the default paths we can auto-select it
+			if len(devcontainers) == 1 && utils.StringInSlice(devcontainers[0].Path, DEFAULT_DEVCONTAINER_DEFINITIONS) {
+				devContainerPath = devcontainers[0].Path
+			} else {
+				promptOptions := []string{}
+
+				if !utils.StringInSlice(devcontainers[0].Path, DEFAULT_DEVCONTAINER_DEFINITIONS) {
+					promptOptions = []string{"Default Codespaces configuration"}
+				}
+
+				for _, devcontainer := range devcontainers {
+					promptOptions = append(promptOptions, devcontainer.Path)
+				}
+
+				devContainerPathQuestion := &survey.Question{
+					Name: "devContainerPath",
+					Prompt: &survey.Select{
+						Message: "Devcontainer definition file:",
+						Options: promptOptions,
+					},
+				}
+
+				if err := ask([]*survey.Question{devContainerPathQuestion}, &devContainerPath); err != nil {
+					return fmt.Errorf("failed to prompt: %w", err)
+				}
+			}
+		}
+
+		if devContainerPath == "Default Codespaces configuration" {
```

Comment:
Since this static string is reused across logic, it could be extracted as a const so that typos can't happen between instances where this is being compared